engine/runtime-parser/src/main/scala/org/enso/compiler/core/ir/Function.scala

```diff
@@ -45,14 +45,25 @@ object Function {
     */
   sealed case class Lambda(
     override val arguments: List[DefinitionArgument],
-    override val body: Expression,
+    val bodyList: Seq[Expression],
     location: Option[IdentifiedLocation],
-    override val canBeTCO: Boolean = true,
-    passData: MetadataStorage      = MetadataStorage(),
-    diagnostics: DiagnosticStorage = DiagnosticStorage()
+    override val canBeTCO: Boolean,
+    passData: MetadataStorage,
+    diagnostics: DiagnosticStorage
   ) extends Function
       with IRKind.Primitive {
+    def this(
+      arguments: List[DefinitionArgument],
+      body: Expression,
+      location: Option[IdentifiedLocation],
+      canBeTCO: Boolean              = true,
+      passData: MetadataStorage      = new MetadataStorage(),
+      diagnostics: DiagnosticStorage = new DiagnosticStorage()
+    ) = {
+      this(arguments, Seq(body), location, canBeTCO, passData, diagnostics)
+    }
     var id: UUID @Identifier = randomId
+    lazy val body            = bodyList.head
```

Comment:
shouldn't this `assert(bodyList.length == 1)`?

Reply:
That check would belong to the constructor that accepts `bodyList`. However as that constructor is generated and I am not sure how to add such check in Scala, it is easier to live without it.